pallets/auction/src/lib.rs

```diff
@@ -956,22 +958,20 @@ pub mod pallet {
 			high_bidder: &T::AccountId,
 			asset_id: &(ClassId, TokenId),
 			social_currency_id: FungibleTokenId,
+			listing_level: ListingLevel<T::AccountId>,
 		) -> DispatchResult {
-			let fee_scale = T::RoyaltyFee::get();
-			// Calculate loyalty fee and deposit to pot fund
-			let royalty_fee = high_bid_price
-				.saturating_mul(fee_scale.into())
-				.checked_div(&10000u128.saturated_into())
-				.ok_or("Overflow")?;
-
-			// Collect loyalty fee
-			// and deposit to class fund
-
 			// Get royalty fee
-			//			let fee = T::NFTHandler::get_nft_detail((asset_id.0,
-			// asset_id.1)).ok_or(Error::<T>::AssetIsNotExist)?;
-
-			let class_fund = T::NFTHandler::get_class_fund(&asset_id.0);
+			let nft_details = T::NFTHandler::get_nft_detail((asset_id.0, asset_id.1))?;
+			let mut royalty_fee = nft_details.royalty_fee * *high_bid_price;
+			let mut class_fund = T::NFTHandler::get_class_fund(&asset_id.0);
+			match listing_level {
+				ListingLevel::Local(metaverse_id) => {
+					class_fund = T::MetaverseInfoSource::get_metaverse_treasury(metaverse_id);
+					royalty_fee =
```

Comment:
royalty_fee and listing fee are 2 different concepts. Royalty fee will go to NFT fund, listing fee will go to metaverse treasury.

Reply:
sure will update the code to support both

Reply:
@justinphamnz  do the global marketplace have a listing fee as well?